libs/navigation/navigation.js

```diff
@@ -35,12 +35,20 @@ function getParamsConfigs(configs) {
 
 export default async function loadBlock(configs, customLib) {
   const {
-    header, footer, authoringPath, env = 'prod', locale = '', theme,
+    header,
+    footer,
+    authoringPath,
+    env = 'prod',
+    locale = '',
+    theme,
+    onReady,
+    onError,
```

Comment:
Since this is an async function couldn't the consumers just do
```javascript
 try {
  // do stuff
  await loadBlock(...);
  // do more stuff
} catch (e) {
  // handle error
}
```
Or they could even do
```javascript
loadBlock(...).then(onReady).catch(onError)
```
of course you'd have to `await` the `initBlock` function inside bootstrapper.js for this to work correctly (https://github.com/adobecom/milo/blob/af281740d8e910d0e677e3fb556f11c7095aeb50/libs/navigation/bootstrapper.js#L31).

This would allow us to avoid the rigmarole of callbacks and events that add more complexity to our model of how the global-navigation works. We'd just have to throw irrecoverable errors for the consumer to deal with (you may have to modify `loadPostLCP` with a `try...catch` to make milo correctly 'consume' global-navigation).
I realize I might be missing some nuance related to what errors should be propagated vs which errors should be handled internally, but my understanding is that the global-navigation already does this adequately. 
Ultimately I think it better to deal with this problem by propagating exceptions and dealing with exceptions on the consuming end rather than muck around with events and callbacks. Some time back we refused to allow Partner Portals to emit events to signify when specific parts of the global-navigation were loaded #2583 , and I think the same concerns apply here.

Reply:
If we implement it this way, the consuming client will receive the onError callback only when the gnav content is not found. 
All other errors, such as IMS issues or errors loading fragments, would be logged internally. To achieve this we could add a console.error line [here](https://github.com/adobecom/milo/blob/6fa95584902286108cfea82572288559b3517bb0/libs/blocks/global-navigation/utilities/utilities.js#L44)
Thoughts? @narcis-radu @mokimo @salonijain3